### PR TITLE
Provide a hook for a plugin handler to receive messages on the web

### PR DESCRIPTION
**Pull request by hterkelsen**
This is the first step to enabling plugins on web: see https://github.com/flutter/flutter/issues/39591

### CODE REVIEW

File: lib/web_ui/lib/src/engine/plugins.dart [hunk 1]
```diff
@@ -0,0 +1,7 @@
+part of engine;
```
Comment (jonahwilliams) [inline lib/web_ui/lib/src/engine/plugins.dart]: Needs license header

File: web_sdk/sdk_rewriter.dart [hunk 2]
```diff
@@ -28,10 +28,9 @@ import 'dart:_engine' as engine;
   <String>[
 r'''
 export 'src/engine.dart'
-    show persistedPictureFactory, houdiniPictureFactory, platformViewRegistry;
 ''',
 r'''
-export 'dart:_engine' show persistedPictureFactory, houdiniPictureFactory, webOnlyInitializeEngine;
+export 'dart:_engine'
```
Comment (jonahwilliams) [inline web_sdk/sdk_rewriter.dart]: I would recommend keeping this export as locked down as possible

Comment (hterkelsen) [reply web_sdk/sdk_rewriter.dart]: Agreed, but I think it's confusing that the exports list is totally overwritten in the rewriter. We should keep the `dart:_engine` exports limited, but the source of truth should be in `lib/ui.dart`

File: lib/web_ui/lib/src/engine/plugins.dart (hunk 1 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/plugins.dart]: Done.

**Pull request merged**